src/jdk.compiler/share/classes/com/sun/tools/javac/code/Types.java

```diff
@@ -2376,7 +2376,7 @@ private boolean implicitIdentityType(Type t) {
                             }
                             break;
                         case MTH:
-                            if ((s.flags() & SYNCHRONIZED) != 0) {
+                            if ((s.flags() & (SYNCHRONIZED | STATIC)) == SYNCHRONIZED) {
                                 return true;
                             } else if (s.isConstructor()) {
```

Comment:
This is a bug fix - static methods may be synchronized or not - immaterial.

Reply:
> Do you think the comment above the signature of `implicitIdentityType` needs to reflect this change in a parenthesis?

Sure, I don't see a comment above implicitIdentityType, but inside it at the top, I have amended it to say "if it declares a synhronized instance method ..."